agent/dns.go

```diff
@@ -1099,15 +1108,16 @@ func rCodeFromError(err error) int {
 	case errors.Is(err, errNameNotFound),
 		structs.IsErrNoDCPath(err),
 		structs.IsErrQueryNotFound(err),
-		structs.IsErrSamenessGroupMustBeDefaultForFailover(err):
+		structs.IsErrSamenessGroupMustBeDefaultForFailover(err),
+		structs.IsErrSamenessGroupNotFound(err):
```

Comment:
FYI, looks like this is a net-new changes based on comparing Ent, but it brings them into sync.